src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AspNet/HttpModule_Integration.cs

```diff
@@ -0,0 +1,63 @@
+#if NETFRAMEWORK
+using System.Threading;
+using System.Web;
+using Datadog.Trace.AspNet;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNet
+{
+    /// <summary>
+    /// System.Web.Compilation.BuildManager.InvokePreStartInitMethodsCore calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "System.Web",
+        TypeName = "System.Web.Compilation.BuildManager",
+        MethodName = "InvokePreStartInitMethodsCore",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { "System.Collections.Generic.ICollection`1[System.Reflection.MethodInfo]", "System.Func`1[System.IDisposable]" },
+        MinimumVersion = "4.0.0",
+        MaximumVersion = "4.*.*",
+        IntegrationName = IntegrationName)]
+    public class HttpModule_Integration
+    {
+        private const string IntegrationName = nameof(IntegrationIds.AspNet);
+
+        /// <summary>
+        /// Indicates whether we're initializing the HttpModule for the first time
+        /// </summary>
+        private static int _firstInitialization = 1;
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TCollection">Type of the collection</typeparam>
+        /// <typeparam name="TFunc">Type of the </typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method. This method is static so this parameter will always be null</param>
+        /// <param name="methods">The methods to be invoked</param>
+        /// <param name="setHostingEnvironmentCultures">The function to set the environment culture</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TCollection, TFunc>(TTarget instance, TCollection methods, TFunc setHostingEnvironmentCultures)
+        {
+            if (Interlocked.Exchange(ref _firstInitialization, 0) != 1)
+            {
+                // The HttpModule was already registered
+                return CallTargetState.GetDefault();
+            }
+
+            try
+            {
+                HttpApplication.RegisterModule(typeof(TracingHttpModule));
```

Comment:
What would be the impact if tracer versions are out of sync (i.e. an older version of the msi with the module registered in config). Do we need to check if the module is already registered first?

If we try and register the module again, would we see two `aspnet.request` spans for every request, or it would it crash?

Reply:
The HttpModule is permanently set (for 1.x versions of the Tracer) to 1.0.0. Perhaps it would more resilient to check if it's already been registered, but our module already has logic to only create one event handler (therefore one span) no matter how many times it's registered: https://github.com/DataDog/dd-trace-dotnet/blob/4541013aa89cc5fc2c8947fb03f28a5854409726/src/Datadog.Trace.AspNet/TracingHttpModule.cs#L62

Reply:
And yes I was able to verify that running the MSI and then also adding the HttpModule to the configuration results in only one `aspnet.request` span